rest-api/src/main/scala/za/co/absa/enceladus/rest_api/controllers/v3/MappingTableControllerV3.scala

```diff
@@ -0,0 +1,88 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.rest_api.controllers.v3
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.{HttpStatus, ResponseEntity}
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.security.core.annotation.AuthenticationPrincipal
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.web.bind.annotation._
+import za.co.absa.enceladus.model._
+import za.co.absa.enceladus.rest_api.services.v3.MappingTableServiceV3
+
+import java.util.concurrent.CompletableFuture
+import javax.servlet.http.HttpServletRequest
+
+@RestController
+@RequestMapping(Array("/api-v3/mapping-tables"))
+class MappingTableControllerV3 @Autowired()(mappingTableService: MappingTableServiceV3)
+  extends VersionedModelControllerV3(mappingTableService) {
+
+  import za.co.absa.enceladus.rest_api.utils.implicits._
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  @GetMapping(path = Array("/{name}/{version}/defaults"))
+  @ResponseStatus(HttpStatus.OK)
+  def getDefaults(@PathVariable name: String,
+                  @PathVariable version: String): CompletableFuture[Seq[DefaultValue]] = {
+
+    forVersionExpression(name, version)(mappingTableService.getVersion).map { // "latest" version is accepted
+      case Some(entity) => entity.defaultMappingValue
+      case None => throw notFound()
+    }
+  }
+
+  @PutMapping(path = Array("/{name}/{version}/defaults"))
+  @ResponseStatus(HttpStatus.CREATED)
+  @PreAuthorize("@authConstants.hasAdminRole(authentication)")
+  def updateDefault(@AuthenticationPrincipal user: UserDetails,
+                 @PathVariable name: String,
+                 @PathVariable version: String,
+                 @RequestBody newDefaults: Array[DefaultValue],
+                 request: HttpServletRequest
+                ): CompletableFuture[ResponseEntity[Validation]] = {
+    for {
+      existingMtOpt <- forVersionExpression(name, version)(mappingTableService.getVersion)
+      existingMt = existingMtOpt.getOrElse(throw notFound())
+      updatedMtAndValidationOpt <- mappingTableService.updateDefaults(user.getUsername, name, existingMt.version, newDefaults.toList)
+      (updatedMt, validation) = updatedMtAndValidationOpt.getOrElse(throw notFound())
+      response = createdWithNameVersionLocationBuilder(name, updatedMt.version, request,
+        stripLastSegments = 3, suffix = s"/defaults").body(validation)  // stripping: /{name}/{version}/defaults
+    } yield response
+  }
+
+  @PostMapping(path = Array("/{name}/{version}/defaults"))
+  @ResponseStatus(HttpStatus.CREATED)
+  @PreAuthorize("@authConstants.hasAdminRole(authentication)")
+  def addDefault(@AuthenticationPrincipal user: UserDetails,
```

Comment:
Post defaults does not validate for duplicities. Is that beyond scope of this PR?

Reply:
Yes, it does not validate that. Incidentally, it seems to work as expected - later additions under the same default would replace the previously added, but I agree this is not ideal. 

(A slight complication is that the underlying data structure is mapped as List, hence it inherently allows duplicates, but it can of course be done).

So what do you think that the validation should be, if any, @Zejnilovic? I see the possibilities as:
1. no validation - keep as is - `List.toMap` will solve the duplicates in favor of the latest
2. maintain the list duplicate-free in V3 API, then how to deal with duplicates:
  a. override silently: `addDefault` with existing key overrides; `replaceDefaults` payload can also contain duplicates - again, we can override 
 b. the same, but with warnings: override, but issue warnings: `addDefault` with existing key overrides (return a warning about duplication override); `replaceDefaults` payload can also contain duplicates - again, we can override and issue validation warning
  c. same for `addDefault` as in (b) - override + validation warning; stricter for `replaceDefaults` - validation error on duplicates
  d. disallow overriding completely: `replaceDefaults` with defaults would be refused - validation error, `addDefault` with a default = validation error, too (would require to run `replaceDefaults` without the key).

My perspective: maybe 2b?

@Zejnilovic @AdrianOlosutean @benedeki

Reply:
Perhaps (1) keep as-is: @benedeki -> consensus seems to be for separate PR.

Reply:
Validations will be a separate issue/pr #2061